src/Modules/Events/EventCardComponent.js

```diff
@@ -26,26 +28,27 @@ const EventCardComponent = props => {
       acceptReservations,
       acceptInterest,
       eventDetails,
-      eventId,
     },
   } = props;
 
+  const redirectToRegistration = (event_date_id) => {
+    history.push(RENDER_URL.EVENT_REGISTRATION_URL, {
+      eventDateId: event_date_id,
+    });
+  };
+
   const ButtonView = () => {
     if (acceptReservations) {
-      return (
-        <LinkContainer to={`${RENDER_URL.EVENT_REGISTRATION_URL}/${id}`}>
-          <button type="button" className="btn custom-button ml-1 flex-grow-1">
-            Reserve Time
-          </button>
-        </LinkContainer>
-      );
+      return <ReserveTimeButton event_date_id={id} />;
     } else if (acceptInterest && !acceptReservations) {
       return (
-        <LinkContainer to={`${RENDER_URL.EVENT_REGISTRATION_URL}/${id}`}>
-          <button type="button" className="btn custom-button ml-1 flex-grow-1">
-            RSVP
-          </button>
-        </LinkContainer>
+        <button
+          type="button"
+          className="btn custom-button ml-1 flex-grow-1"
+          onClick={() => redirectToRegistration(id)}
```

Comment:
This is an anti pattern. Changing the route via directly using `history` breaks a11y. Best to keep the Link Container.